daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -831,6 +831,20 @@ private[lf] object Speedy {
     ): Machine =
       fromPureSExpr(compiledPackages, compiledPackages.compiler.unsafeCompile(expr))
 
+    @throws[PackageNotFound]
+    @throws[CompilationError]
+    // Construct a machine for testing.
+    private[speedy] def dummy(
+        compiledPackages: CompiledPackages,
+    ): Machine = Machine(
+      compiledPackages = compiledPackages,
+      submissionTime = Time.Timestamp.MinValue,
+      initialSeeding = InitialSeeding.NoSeed,
+      expr = null,
+      globalCids = Set.empty,
+      committers = Set.empty,
```

Comment:
is it really worthwhile to create a new builder if you use it only once ?

Reply:
I can remove it, I was hoping it was useful.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8272/commits/bda089345a2d7c6d7d9dbac52a2374a6e5d2390a